src/vs/workbench/api/common/extHostNotebookDocument.ts

```diff
@@ -110,6 +110,14 @@ export class ExtHostCell {
 				output.items.length = 0;
 			}
 			output.items.push(...newItems);
+			if (output.items.every(item => notebookCommon.isTextStreamMime(item.mime))) {
+				const compressed = notebookCommon.compressOutputItemStreams(newOutputItems[0].mime, newOutputItems.map(item => item.valueBytes));
```

Comment:
I thought this was going to be done in the renderer process

Reply:
I thought this is what we discussed.

Yes we do this in the renderer process, but when we get the event back from renderer process in extnesion host we need to compress the output on the round trip. this way the serializers have the compressed outputs. 
Else serializers also have to perform the compression.
If we compress in renderer process and send the compressed output, then if there are no `\r` characters or the like, then the output keeps growing and we end up always sending a very large string from renderer back to the exension host via the change events.

Hope that makes sense.